harness/determined/common/api/request.py

```diff
@@ -356,3 +363,37 @@ def ws(host: str, path: str) -> WebSocket:
     token = authentication.must_cli_auth().get_session_token()
     websocket.add_header("Authorization".encode(), "Bearer {}".format(token).encode())
     return WebSocket(websocket)
+
+
+def parse_warnings(
+    warnings: Optional[Union[Sequence[bindings.v1LaunchWarning], Sequence[int]]]
+) -> Optional[Sequence[bindings.v1LaunchWarning]]:
+
+    # Warnings may be passed in as either an enum value or integer value
+    warning_message_int_to_enum_value = {
+        1: bindings.v1LaunchWarning.LAUNCH_WARNING_CURRENT_SLOTS_EXCEEDED
+    }
+
+    warning_enum_values: list[bindings.v1LaunchWarning] = []
+    if warnings:
+        for warning in warnings:
+            if isinstance(warning, int):
+                warning_value: int = warning
+                warning_enum_values.append(warning_message_int_to_enum_value[warning_value])
+            elif isinstance(warning, str):
+                warning_enum_value: bindings.v1LaunchWarning = bindings.v1LaunchWarning(warning)
+                warning_enum_values.append(warning_enum_value)
+
+    return warning_enum_values
+
+
+def handle_warnings(warnings: Optional[Sequence[bindings.v1LaunchWarning]]) -> None:
+    warning_message_map = {
+        bindings.v1LaunchWarning.LAUNCH_WARNING_CURRENT_SLOTS_EXCEEDED: (
+            "Warning: The requested job requires more slots than currently available. "
+            "You may need to increase cluster resources in order for the job to run."
+        )
+    }
+    if warnings:
+        for warning in warnings:
+            print(colored(warning_message_map[warning], "yellow"))
```

Comment:
Generally, you don't want to print warnings to `stdout` in a CLI because `stdout` is commonly used in bash pipelines.  You don't want warnings to break those pipelines, and you do want the warnings to show even when a pipeline is in use.

Easy fix though, just print to `file=sys.stderr` instead.

Reply:
updated